.github/workflows/pkgci_build_packages.yml

```
@@ -0,0 +1,124 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: PkgCI Build Packages
+on:
+  workflow_call:
+    inputs:
+      package_version:
+        type: string
+        required: true
+
+jobs:
+  linux_x86_64_release_packages:
+    name: Linux Release Packages (x86_64)
+    runs-on:
+      - self-hosted # must come first
+      - runner-group=${{ github.event_name == 'pull_request' && 'presubmit' || 'postsubmit' }}
+      - environment=prod
+      - cpu
+      - os-family=Linux
+    strategy:
+      fail-fast: false
+    env:
+      CACHE_DIR: ${{ github.workspace }}/.iree-container-cache
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3 # v3.5.0
+        with:
+          submodules: true
+      - name: Write version info
+        shell: bash
+        run: |
+          cat << EOF > version_info.json
+          {
+            "package-suffix": "",
+            "package-version": "${{ inputs.package_version }}",
+            "iree-revision": "$(cd ../iree && git rev-parse HEAD)",
+          }
+          EOF
+          realpath version_info.json
+          cat version_info.json
+      - name: Enable cache
+        uses: actions/cache@v3
```

We typically pin actions to a git hash rather than a tag, so someone can't change the code out from under us (bit of a buffer against compromised packages... if there is an issue, it is hopefully detected and fixed before we choose to update)

```suggestion
      - name: Enable cache
        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
```

(Yes, this comes from a Google security policy: https://opensource.google/documentation/reference/github/services#actions, which I agree with. No, it isn't enforced and I do see it as less convenient.)

---

BTW, I found that the GitHub Actions caching was quite limiting on both storage size limits and upload/download speed, even after tuning compression settings and pruning directory payload sizes. If it works well enough now / for this, great. Otherwise, we may want to use a cloud provider to store the cached files and/or persist cache storage on the runners.

---

We've been using it in downstreams reasonably well for quite some time, and part of the thing that has made IREE "weird" for those of us who work on both is that it has so much diverted plumbing on this stuff. I don't know if things got better or there is a different bar for "good enough", but if possible, it is a lot better to have things that work well for both upstream/downstreams.